tests/linearizability/model.go

```diff
@@ -93,121 +91,82 @@ var etcdModel = porcupine.Model{
 }
 
 func step(state EtcdState, request EtcdRequest, response EtcdResponse) (bool, EtcdState) {
-	if request.Key == "" {
-		panic("invalid request")
-	}
-	if state.Key == "" {
-		return true, initState(request, response)
+	if len(state.PossibleValues) == 0 {
+		state.Key = request.Key
+		if ok, v := initValueRevision(request, response); ok {
+			state.PossibleValues = append(state.PossibleValues, v)
+		}
+		return true, state
 	}
 	if state.Key != request.Key {
-		panic("Multiple keys not supported")
+		panic("multiple keys not supported")
 	}
-	switch request.Op {
-	case Get:
-		return stepGet(state, request, response)
-	case Put:
-		return stepPut(state, request, response)
-	case Delete:
-		return stepDelete(state, request, response)
-	default:
-		panic("Unknown operation")
+	if response.Err != nil {
+		for _, kv := range state.PossibleValues {
+			kv.handle(request)
+			state.PossibleValues = append(state.PossibleValues, kv)
+		}
+	} else {
+		var i = 0
+		for _, kv := range state.PossibleValues {
+			expectedResponse := kv.handle(request)
+			if expectedResponse == response {
+				state.PossibleValues[i] = kv
+				i++
+			}
+		}
+		state.PossibleValues = state.PossibleValues[:i]
 	}
+	return len(state.PossibleValues) > 0, state
+}
+
+type valueRevision struct {
```

Comment:
Please put `valueRevision` at the place either close to its method `(*valueRevision) handle` or where it is used `PossibleValues []valueRevision`